resources/indexer/indexer.py

```diff
@@ -670,14 +635,140 @@ def gather_file_info(dirs, site, basic_only=False):
+        if path.startswith(root):
+            return
+    message = f"{path} is not rooted at: {', '.join(ACCEPTED_ROOTS)}"
+    logging.critical(message)
+    raise Exception(f'Invalid path ({message}).')
+
+
+def gather_file_info(starting_paths, blacklist, args):
+    """Gather and post metadata from files rooted at starting_paths. Do this multi-processed."""
+    # Get full paths
+    starting_paths = [os.path.abspath(p) for p in starting_paths]
+    for p in starting_paths:
+        check_path(p)
+
+    # Traverse paths and process files
+    futures = []
+    with ProcessPoolExecutor() as pool:
+        queue = starting_paths
+        split = math.ceil(len(queue) / args.processes)
+        while futures or queue:
+            logging.debug(f'Queue: {len(queue)}.')
+            # Divvy up queue among available worker(s). Each worker gets 1/nth of the queue.
+            if queue:
+                queue.sort()
+                while args.processes != len(futures):
+                    paths, queue = queue[:split], queue[split:]
+                    logging.debug(f'Worker Assigned: {len(futures)+1}/{args.processes} ({len(paths)} paths).')
+                    futures.append(pool.submit(process_work, paths, args, blacklist))
+            logging.debug(f'Workers: {len(futures)} {futures}.')
+            # Extend the queue
+            while not futures[0].done():  # concurrent.futures.wait(FIRST_COMPLETED) is slower
+                sleep(0.1)
+            future = futures.pop(0)
+            result = future.result()
+            if result:
+                queue.extend(result)
+                split = math.ceil(len(queue) / args.processes)
+            logging.debug(f'Worker finished: {future} (enqueued {len(result)}).')
+
+
+def sorted_unique(infile, others=None):
+    """Read in lines from file, aggregate with those in others list. Return all unique lines, sorted."""
+    lines = []
+    if others:
+        lines.extend(others)
+    if infile:
+        with open(infile) as file:
+            lines.extend([l.rstrip() for l in file])
+    lines = [l for l in sorted(set(lines)) if l]
+    return lines
+
+
+def main():
+    """Main."""
     parser = argparse.ArgumentParser(description='Find files under PATH(s), compute their metadata and '
                                      'upload it to File Catalog.',
                                      epilog='Notes: (1) symbolic links are never followed.',
                                      formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-    parser.add_argument('path', metavar='PATH', nargs='+',
+    parser.add_argument('paths', metavar='PATHS', nargs='*',
                         help='path(s) to scan for files.')
+    parser.add_argument('--paths-file', dest='paths_file', default=None,
+                        help='file containing path(s) to scan for files. (use this option for a large number of paths)')
+    parser.add_argument('--processes', type=int, default=1,
```

Comment:
Since it's processing in condor, a default of 1 is actually fairly good.  You could technically look in the condor classad for the number of cpus allocated.

Asking for the number of cpus on the machine is probably the worst approach when running in condor.

Reply:
As @dsschult touched on, this is meant to be defined by the parent/condor script that runs it